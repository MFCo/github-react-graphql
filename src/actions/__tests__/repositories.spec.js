import configureStore from 'redux-mock-store';
import act from '../../actions';

const middlewares = [];
const mockStore = configureStore(middlewares);

it('should dispatch action to add repositories', () => {

  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(act.updateRepositories(
    {
      newList:
        [
          {
            "name": "PangRevenge",
            "description": "Pang pang revenge",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
          {
            "name": "PangRevenge",
            "description": "Pangolier is balanced ",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
          {
            "name": "PangRevenge",
            "description": "Yeah, i love dota2 ",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
        ],
      newHasNextPage: true,
      newEndCursor: "Y3Vyc29yOnYyOpHOARWdng=="
    }));
  const actions = store.getActions();
  expect(actions).toMatchSnapshot();
});

it('should dispatch action to add repositories', () => {

  const initialState = {
    list:
      [
        {
          "name": "PangRevenge",
          "description": "Pang pang revenge",
          "forkCount": 0,
          "primaryLanguage": {
            "color": "#f1e05a",
            "name": "JavaScript"
          },
          "homepageUrl": null
        },
        {
          "name": "PangRevenge",
          "description": "Pangolier is balanced ",
          "forkCount": 0,
          "primaryLanguage": {
            "color": "#f1e05a",
            "name": "JavaScript"
          },
          "homepageUrl": null
        },
        {
          "name": "PangRevenge",
          "description": "Yeah, i love dota2 ",
          "forkCount": 0,
          "primaryLanguage": {
            "color": "#f1e05a",
            "name": "JavaScript"
          },
          "homepageUrl": null
        }
      ],
    hasNextPage: true,
    endCursor: "Y3Vyc29yOnYyOpHOARWdng=="
  };

  const store = mockStore(initialState);

  store.dispatch(act.addRepositories(
    {
      newList:
        [
          {
            "name": "PangRevenge",
            "description": "We should play",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
          {
            "name": "PangRevenge",
            "description": "Hahaha",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          }
        ],
      newHasNextPage: false,
      newEndCursor: "Y3Vyc29yOnYyOpHOARWdng=="
    }));
  const actions = store.getActions();
  expect(actions).toMatchSnapshot();
});