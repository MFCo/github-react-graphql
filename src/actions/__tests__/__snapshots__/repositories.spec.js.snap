// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should dispatch action to add repositories 1`] = `
Array [
  Object {
    "payload": Object {
      "newEndCursor": "Y3Vyc29yOnYyOpHOARWdng==",
      "newHasNextPage": true,
      "newList": Array [
        Object {
          "description": "Pang pang revenge",
          "forkCount": 0,
          "homepageUrl": null,
          "name": "PangRevenge",
          "primaryLanguage": Object {
            "color": "#f1e05a",
            "name": "JavaScript",
          },
        },
        Object {
          "description": "Pangolier is balanced ",
          "forkCount": 0,
          "homepageUrl": null,
          "name": "PangRevenge",
          "primaryLanguage": Object {
            "color": "#f1e05a",
            "name": "JavaScript",
          },
        },
        Object {
          "description": "Yeah, i love dota2 ",
          "forkCount": 0,
          "homepageUrl": null,
          "name": "PangRevenge",
          "primaryLanguage": Object {
            "color": "#f1e05a",
            "name": "JavaScript",
          },
        },
      ],
    },
    "type": "UPDATE_REPOSITORIES",
  },
]
`;

exports[`should dispatch action to add repositories 2`] = `
Array [
  Object {
    "payload": Object {
      "newEndCursor": "Y3Vyc29yOnYyOpHOARWdng==",
      "newHasNextPage": false,
      "newList": Array [
        Object {
          "description": "We should play",
          "forkCount": 0,
          "homepageUrl": null,
          "name": "PangRevenge",
          "primaryLanguage": Object {
            "color": "#f1e05a",
            "name": "JavaScript",
          },
        },
        Object {
          "description": "Hahaha",
          "forkCount": 0,
          "homepageUrl": null,
          "name": "PangRevenge",
          "primaryLanguage": Object {
            "color": "#f1e05a",
            "name": "JavaScript",
          },
        },
      ],
    },
    "type": "ADD_REPOSITORIES",
  },
]
`;
